fdm-core/src/field.ts

```diff
@@ -425,3 +426,185 @@ export async function updateField(
         }
     })
 }
+
+/**
+ * Removes a field and all its associated data.
+ *
+ * This function checks if the principal has permission to delete the field, then proceeds to delete
+ * the field and all cascading data, including acquiring and discarding information, cultivations,
+ * fertilizer applications, soil analyses, and harvests.
+ *
+ * @param fdm The FDM instance providing the connection to the database. The instance can be created with {@link createFdmServer}.
+ * @param principal_id - The unique identifier of the principal performing the operation.
+ * @param b_id - The unique identifier of the field to be removed.
+ * @returns A promise that resolves when the field has been successfully removed.
+ *
+ * @throws {Error} If the principal does not have permission to delete the field.
+ *
+ * @alpha
+ */
+export async function removeField(
+    fdm: FdmType,
+    principal_id: PrincipalId,
+    b_id: schema.fieldsTypeSelect["b_id"],
+): Promise<void> {
+    try {
+        await checkPermission(
+            fdm,
+            "field",
+            "write",
+            b_id,
+            principal_id,
+            "removeField",
+        )
+
+        await fdm.transaction(async (tx: FdmType) => {
+            // Step 1: Get all cultivation IDs for the given field
+            const cultivations = await tx
```

Comment:
if you remove the field, then data from the past is also deleted isn't?
this is another "action" than the usage of a "field", where a farmer can use the field from year A to year B. In that case the data remain stored. 
check: do we permanently delete all data, or can we reset the situation if someone regrets the decision made to delete one.